python/dgl/transform.py

```diff
@@ -575,6 +575,45 @@ def partition_graph_with_halo(g, node_part, num_hops):
         subg_dict[i] = subg
     return subg_dict
 
+def metis_partition(g, k, num_hops=1):
+    '''
+    This is to partition a graph with Metis partitioning.
+
+    The partitioned graph is stored in DGLGraph. The DGLGraph has the `inner_node` node data
```

Comment:
I feel when we say a graph is partitioned, we mean both its nodes and edges are partitioned. It looks like Metis partition only assigns nodes to partitions, so it leaves the edge assignment to our discretion. If that's the case, I think maybe the following description is better?
* The default behavior is storing in-edges to the nodes' partition.
    - The `inner_node` and `inner_edge` arrays are all one in this case.
    - You have the option to not even include them.
* Rename `num_hops` to `extra_cached_hops` and make its default be 0.
* Consider merge `inner_node` and `inner_edge` to `part_id` because for the i^th partition, the user could just use `th.nonzero(g.ndata['part_id'] == i)` to get the boolean array.

All these assume that the partitioning results in disjoint assignment, so if your actual implementation adopts node/edge mirror storage, the semantic will break.